src/common.h

```diff
@@ -184,4 +188,56 @@ Join(const T& container, const std::string& delim)
   return ss.str();
 }
 
+
+// Used by Python Bindings to accept arguments to initialize Frontends.
+// Known pybind11 issue: bool has to come before int for std::variant
+using VariantType = std::variant<bool, int, std::string>;
+using UnorderedMapType = std::unordered_map<std::string, VariantType>;
+
+
+template <typename T>
+TRITONSERVER_Error*
+get_value(const UnorderedMapType& options, const std::string& key, T* arg)
+{
+  auto curr = options.find(key);
+  bool is_present = (curr != options.end());
+  std::string msg;
+
+  if (!is_present) {
+    msg = "Key: " + key + " not found in options provided.";
+    return TRITONSERVER_ErrorNew(TRITONSERVER_ERROR_INVALID_ARG, msg.c_str());
+  }
+
+  bool correct_type = std::holds_alternative<T>(curr->second);
+  if (!correct_type) {
+    std::string expected;
+    std::string found;
+    VariantType value = *arg;
+    if (std::holds_alternative<int>(value)) {
+      expected = "int";
+    } else if (std::holds_alternative<bool>(value)) {
+      expected = "bool";
+    } else if (std::holds_alternative<std::string>(value)) {
+      expected = "string";
+    }
```

Comment:
@GuanLuo 
The reason I strayed from doing `typeid(*arg).name()` is because doing so outputs the mangled data type. 

For example, if `arg` is of type `std::string`, the value of `expected` will be something along the lines of `NSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE`